vllm/engine/metrics.py

```diff
@@ -60,6 +60,22 @@ def __init__(self, labelnames: List[str]):
             name="vllm:generation_tokens_total",
             documentation="Number of generation tokens processed.",
             labelnames=labelnames)
+        self.counter_request_success = Counter(
+            name="vllm:request_success",
+            documentation="Count of successfully processed requests.",
+            labelnames=labelnames)
+        self.histogram_request_prompt_tokens = Histogram(
+            name="vllm:request_prompt_tokens",
+            documentation="Number of prefill tokens processed.",
+            labelnames=labelnames,
+            buckets=build_1_2_5_buckets(max_model_len),
+        )
+        self.histogram_request_generation_tokens = Histogram(
+            name="vllm:request_generation_tokens",
+            documentation="Number of generation tokens processed.",
+            labelnames=labelnames,
+            buckets=build_1_2_5_buckets(max_model_len),
+        )
```

Comment:
Thanks for your feedback!
You are right. But, wouldn't it be beneficial to have in addition histograms depicting the distribution of prompt length and generation length?

Reply:
Since this metric doesn't actually introduce any overhead (because the data from `vllm:x_tokens_total` is reused, these two are probably fine. It would be interesting to know how big the prompts the users were providing are.

Reply:
Exactly! I suggest to deprecate the 2 `vllm:x_tokens_total` metrics as they will be included as part of the Histogram metrics this PR adds.

Reply:
I think we should keep these metrics, because a developer may not want to have to aggregate histogram data in order to get the same effect of `vllm:x_tokens_total`

Reply:
Prometheus histograms have this nice feature where in addition to the bucket counters, they include 2 additional counters 
suffixed with `_sum` and `_count`.

`_count` is incremented by 1 on every observe, and `_sum` is incremented by the value of the observation.

Therfore, `vllm:prompt_tokens_total` is equivalent to `vllm:request_prompt_tokens_sum`,
and `vllm:generation_tokens_total` is equivalent to `vllm:request_generation_tokens_sum`

Source:
https://www.robustperception.io/how-does-a-prometheus-histogram-work/

Reply:
Oh I see, thanks for explaining. In that case you could move the `vllm:x_tokens_total` metrics into the `# Legacy metrics` section.

Although I think there might be some objection to changing metrics that people are already using in dashboards.

cc @simon-mo @Yard1

Reply:
from my point of view, it's fine to duplicate metrics for backward compatibility reason.

Reply:
Sure, I'll relocate these metrics to the legacy section. Perhaps in the future, when we're able to make breaking changes, we can consider removing them.